client-local/src/main/java/com/cloudera/livy/client/local/LocalClient.java

```diff
@@ -372,18 +369,30 @@ public void run() {
         String allJars = Joiner.on(",").join(livyJars, userJars);
         conf.set(SPARK_JARS_KEY, allJars);
       } else {
-        argv.add("--jars");
-        argv.add(livyJars);
+        argv.put("jars",livyJars);
       }
 
-      argv.add(jar);
-      argv.add("--remote-host");
-      argv.add(serverAddress);
-      argv.add("--remote-port");
-      argv.add(serverPort);
-
-      LOG.info("Running client driver with argv: {}", Joiner.on(" ").join(argv));
-      final Process child = new ProcessBuilder(argv.toArray(new String[argv.size()])).start();
+      //LOG.info("Running client driver with argv: {}", Joiner.on(" ").join(argv));
+      //-----Comment existing livy launcher
+     // final Process child = new ProcessBuilder(argv.toArray(new String[argv.size()])).start();
+        final SparkLauncher launcher= new SparkLauncher();
+        launcher.setSparkHome(argv.get("sparkHome"));
+        launcher.setAppResource(jar);
+        launcher.setAppName("ClientID");
+        launcher.setMainClass(argv.get("class"));
+        launcher.setMaster(master);
+        launcher.setConf("spark.driver.memory",argv.get("DriverMemory"));
+        launcher.setConf("spark.executor.cores",argv.get("executor-cores"));
+        launcher.setConf("spark.executor.memory",argv.get("executor-memory"));
+        launcher.setConf("spark.executor.instances",argv.get("num-executors"));
+        launcher.setConf("spark.driver.extraClassPath",argv.get("ClassPath"));
+        launcher.setConf("spark.driver.extraLibraryPath",argv.get("LibraryPath"));
+        for(String filepath : jars) { launcher.addJar(filepath); }
+        launcher.addAppArgs("--properties-file" + properties.getAbsolutePath());
+        launcher.addAppArgs("--remote-host" + serverAddress);
+        launcher.addAppArgs("--remote-port" + serverPort);
+
+        final Process child = launcher.launch();
```

Comment:
This is fine for now, especially since it doesn't require changing the Spark version (this particular API was added in 1.4). But eventually we want to actually move to 1.6 and use `launcher.startApplication()` instead, which gives us better control and more information about the running Spark app.